File: week3/powx-n.java
class Solution {
    public double myPow(double x, int n) {
        return myPow(x, n, 1);
    }
    
    public double myPow(double x, int n, double a) {
        if (n == 0) {
            return a;
        } else if (n > 0) {
            return myPow(x, n - 1, a * x);
        } else {
            return myPow(x, n + 1, a / x);
        }
    }
}

File: 2026/week3/powx-n.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        if n < 0:
            x = 1 / x
            n = -n
            
        return self.helper(x, n)

    def helper(self, x: float, n: int) -> float:
        if n == 0:
            return 1
        
        res = self.helper(x, n // 2)
        
        if n % 2 == 0:
            return res * res
        else:
            return x * res * res
